### PR TITLE
SAI-Challenger test framework and test cases - round 2

### PR DESCRIPTION
**Pull request by chrispsommers**
This replaces https://github.com/Azure/DASH/pull/259 with cleanups and clarifications.
- Introduced a new way to develop tests - DASH-like test configuration used for both DUT and traffic generator.
- Added scalable test examples for VNET2VNET use case with SAI configuration generation using “dpugen.saigen” module.
- Added OTG/snappi helper modules to scale traffic profile.
- Added SAI-Challenger submodule and testbed definitions. A dev branch is being used, merging to SAI master will occur in the near future and DASH will switch to master accordingly.
- Extended existing dash-saithrift-client with SAI-Challenger client modules and dpugen requirements.
- Enabled new test cases on CI
- Updated snappi/Ixia-C versions
- Documentation
- Fix https://github.com/Azure/DASH/issues/235 by adding more switch attributes to bmv2 libsai

### CODE REVIEW

File: dash-pipeline/tests/saithrift/pytest/switch/test_saithrift_switch.py [hunk 1]
```diff
@@ -6,10 +6,32 @@
        
 @pytest.mark.saithrift
 @pytest.mark.bmv2
-def test_sai_thrift_get_switch_attribute(saithrift_client):
```
Comment (lguohan) [inline dash-pipeline/tests/saithrift/pytest/switch/test_saithrift_switch.py:9]: this is a fix, we should do it in a separate pr.

Comment (chrispsommers) [reply dash-pipeline/tests/saithrift/pytest/switch/test_saithrift_switch.py:9]: @lguohan we can do that, thanks

**Comment by chrispsommers** [comment]
Following today's WG meeting, we will switch to draft status and make a few minor adjustments. Please feel free to continue to post review feedback.

File: dash-pipeline/tests/saithrift/pytest/switch/test_saithrift_switch.py (hunk 1 above)
Comment (chrispsommers) [reply dash-pipeline/tests/saithrift/pytest/switch/test_saithrift_switch.py:9]: @lguohan I created https://github.com/Azure/DASH/pull/281 to do this separately. Once #281 is merged, this PR 278 will be synced and get smaller. Thanks for suggestion.

Comment (chrispsommers) [reply dash-pipeline/tests/saithrift/pytest/switch/test_saithrift_switch.py:9]: Update - once https://github.com/Azure/DASH/pull/281 is merged, we'll sync this PR to main and some of the changes will disappear.

File: dash-pipeline/SAI/templates/utils.cpp.j2 [hunk 2]
```diff
@@ -217,20 +217,77 @@ sai_status_t sai_create_switch_dummy(
     return SAI_STATUS_SUCCESS;
 }
 
-sai_status_t sai_get_switch_attribute_dummy(
+#define DASH_BMV2_NUM_PORTS 2
```
Comment (lguohan) [inline dash-pipeline/SAI/templates/utils.cpp.j2:220]: is this duplicate with pr #281

Comment (chrispsommers) [reply dash-pipeline/SAI/templates/utils.cpp.j2:220]: > is this duplicate with pr #281 ?

Yes it is, see https://github.com/Azure/DASH/pull/278#discussion_r1020607888. Once 281 is merged, we'll sync this PR and some changes will disappear because they'll already be in main.

**Comment by chrispsommers** [comment]
Superceded by https://github.com/Azure/DASH/pull/288
